mcapi/src/v1_15_2/java/net/flintmc/mcapi/v1_15_2/chat/component/event/VersionedOpenUrlActionExecutor.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * FlintMC
+ * Copyright (C) 2020-2021 LabyMedia GmbH and contributors
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package net.flintmc.mcapi.v1_15_2.chat.component.event;
+
+import com.google.common.collect.Sets;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Set;
+import net.flintmc.framework.inject.logging.InjectLogger;
+import net.flintmc.mcapi.chat.component.event.ClickActionExecutor;
+import net.flintmc.mcapi.chat.component.event.ClickEvent;
+import net.flintmc.mcapi.chat.component.event.ClickEvent.Action;
+import net.flintmc.mcapi.chat.component.event.ClickEventAction;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.screen.ConfirmOpenLinkScreen;
+import net.minecraft.client.gui.screen.Screen;
+import net.minecraft.util.Util;
+import org.apache.logging.log4j.Logger;
+
+@Singleton
+@ClickEventAction(value = Action.OPEN_URL, priority = 0)
+public class VersionedOpenUrlActionExecutor implements ClickActionExecutor {
+
+  private static final Set<String> ALLOWED_PROTOCOLS = Sets.newHashSet("http", "https");
```

Comment:
Maybe we could allow adding additional protocols which can then be handled by packages?

Reply:
Decided to be done later